outbox/src/main/kotlin/com/ritense/outbox/OutboxService.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.outbox
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.ObjectNode
+import mu.KotlinLogging
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+open class OutboxService(
+    private val outboxMessageRepository: OutboxMessageRepository,
+    private val objectMapper: ObjectMapper,
+) {
+
+    @Transactional(propagation = Propagation.MANDATORY)
+    open fun send(message: Any) {
```

Comment:
I don't believe we should support sending a message without providing a specific type. Moving classes will affect the type.

Reply:
Why not? The function is here because 9 out of 10 times the message type will be the class name. 

Also, moving classes won't affect the type

Reply:
After looking into it, the eventType won't be needed in the outbox table. This will be inside the BaseEvent which will be saved inside the outbox.message JSON

Reply:
I feel like that type should be a base event in stead of any though. We want to guarenteee that there are a few properties in the message that is sent so that consumers can filter based on it. Issue is that the event is only introduced in the next PR.

Reply:
Yes, the base-event is introduced in the next PR. In that PR the `Any` here is changed to the base event